test/functional/Java17andUp/src_170/org/openj9/test/jep389/downcall/DuplicateStructTests.java

```diff
@@ -0,0 +1,2436 @@
+			Assert.assertEquals(MemoryAccess.getLongAtOffset(resultSegmt, 0),  988765433098L);
+			Assert.assertEquals(MemoryAccess.getLongAtOffset(resultSegmt, 8),  125679011345L);
+			Assert.assertEquals(MemoryAccess.getLongAtOffset(resultSegmt, 16), 126790122456L);
+		}
+	}
+
+	@Test
+	public void test_addNestedLongStructArrayStructs_dupStruct() throws Throwable {
+		GroupLayout longStruct = MemoryLayout.structLayout(longLayout.withName("elem1"), longLayout.withName("elem2"));
+		SequenceLayout structArray = MemoryLayout.sequenceLayout(2, longStruct);
+		GroupLayout structLayout = MemoryLayout.structLayout(structArray.withName("struct_array_elem1"), longLayout.withName("elem2"));
+		FunctionDescriptor fd = FunctionDescriptor.of(structLayout, structLayout, structLayout);
+		Addressable functionSymbol = nativeLibLookup.lookup("addNestedLongStructArrayStructs_dupStruct").get();
+
+		try (ResourceScope scope = ResourceScope.newConfinedScope()) {
+			SegmentAllocator allocator = SegmentAllocator.ofScope(scope);
+			MethodHandle mh = clinker.downcallHandle(functionSymbol, allocator, mt, fd);
+
+			MemorySegment structSegmt1 = allocator.allocate(structLayout);
+			MemoryAccess.setLongAtOffset(structSegmt1, 0, 111111111L);
+			MemoryAccess.setLongAtOffset(structSegmt1, 8, 222222222L);
+			MemoryAccess.setLongAtOffset(structSegmt1, 16, 333333333L);
+			MemoryAccess.setLongAtOffset(structSegmt1, 24, 244444444L);
+			MemoryAccess.setLongAtOffset(structSegmt1, 32, 155555555L);
+			MemorySegment structSegmt2 = allocator.allocate(structLayout);
+			MemoryAccess.setLongAtOffset(structSegmt2, 0, 111111111L);
+			MemoryAccess.setLongAtOffset(structSegmt2, 8, 222222222L);
+			MemoryAccess.setLongAtOffset(structSegmt2, 16, 333333333L);
+			MemoryAccess.setLongAtOffset(structSegmt2, 24, 144444444L);
+			MemoryAccess.setLongAtOffset(structSegmt2, 32, 255555555L);
+
+			MemorySegment resultSegmt = (MemorySegment)mh.invokeExact(structSegmt1, structSegmt2);
+			Assert.assertEquals(MemoryAccess.getLongAtOffset(resultSegmt, 0),  222222222L);
+			Assert.assertEquals(MemoryAccess.getLongAtOffset(resultSegmt, 8),  444444444L);
+			Assert.assertEquals(MemoryAccess.getLongAtOffset(resultSegmt, 16), 666666666L);
+			Assert.assertEquals(MemoryAccess.getLongAtOffset(resultSegmt, 24), 388888888L);
+			Assert.assertEquals(MemoryAccess.getLongAtOffset(resultSegmt, 32), 411111110L);
+		}
+	}
+
+	@Test
+	public void test_addLongStruct1AndNestedLongStructArrayStruct2_returnStruct1_dupStruct() throws Throwable {
+		GroupLayout structLayout1 = MemoryLayout.structLayout(longLayout.withName("elem1"), longLayout.withName("elem2"));
+		VarHandle longHandle1 = structLayout1.varHandle(long.class ,PathElement.groupElement("elem1"));
+		VarHandle longHandle2 = structLayout1.varHandle(long.class ,PathElement.groupElement("elem2"));
+
+		GroupLayout longStruct = MemoryLayout.structLayout(longLayout.withName("elem1"), longLayout.withName("elem2"));
```

Comment:
This local is unused, please remove it. Also the copy on line 1818.

Reply:
Fixed.